feat: Update Event Retention doc (#1460)
edgexfoundry/edgex-docs: docs_src/microservices/core/data/details/DataRetention.md

* `MinCap` is the minimum capacity of the event, the total count of event should be kept in Core Data after purging. you can purge all event if set `MinCap` = `-1`
* `Duration` is the duration to keep the event, the expired events should be detected for purging, and the service can keep old events if `MinCap` is not `-1`, you can disable the expired checking if set `duration` to "0s". Valid time units are "s", "m", "h", e.g., "1.5h" or "2h45m".

!!! note

>     Core Data applies the default retention policy to events not generated by auto events, since some device services (e.g., device-rest, device-mqtt-go) lack the concept of auto events. Thus, the policy must cover these events as well.

Reviewer: > 

The suggested change above is incorrect. Please note that device-rest and device-mqtt-go **DO NOT** support the AutoEvent concept.

My suggestion:
```suggestion
    For events generated by device services that do not support the AutoEvent concept—such as device-rest and device-mqtt-go—Core Data applies the default retention policy to handle these events.
```
Additionally, I feel that this note would be more appropriately placed under the section below.
